qiskit/pulse/parameter_manager.py

```diff
@@ -300,18 +300,19 @@ def visit_ScheduleBlock(self, node: ScheduleBlock):
 
         .. note:: ``ScheduleBlock`` can have parameters in blocks and its alignment.
         """
-        for parameter in node.parameters:
-            self.parameters.add(parameter)
+        # Note that node.parameters returns parameters of main program with subroutines.
+        # The manager of main program is not aware of parameters in subroutines.
+        self.parameters = self.parameters | node._parameter_manager.parameters
```

Comment:
Very minor, but here and elsewhere, `self.parameters |= node._parameter_manager.parameters` is more performant (if it's safe) because it does in-place mutation without creating a new object.

Reply:
Thanks done in https://github.com/Qiskit/qiskit-terra/pull/8005/commits/329c6bc386e81c8a73f6da60e64d3bc47a492f9e